lib/parse.js

```diff
@@ -3,14 +3,14 @@
  * will be expanded greatly in the future in order to support the Scratch 1.4
  * file format. For now, this is nothing but an (awkward) async wrapper around
  * the `JSON.parse` function.
- *
- * @param {String} Input
- *
- * @return {Object}
+ * @param {string}   input    Stringified JSON object
+ * @param {Function} callback Returns error or parsed JSON object
+ * @return {void}
  */
 module.exports = function (input, callback) {
     try {
-        callback(null, JSON.parse(input));
+        var result = JSON.parse(input);
+        callback(null, result);
```

Comment:
I meant to change what we previously had earlier, but didn't want to do it in the pther PR because it was already so giant. I think we should move the callback outside the try catch. I think this was causing an issue, which @rschamp and I had a brief conversation about two weeks ago, with async.waterfall where the callback function has the potential to get called multiple times if the JSON.parse line passes but the callback throws an error.

Reply:
👍

Reply:
Yep, I think moving it into its own statement should help with the particular problem we were seeing. And yeah, I think it would be a good test case to add. We can open an issue for that since we have other tests to add as well.